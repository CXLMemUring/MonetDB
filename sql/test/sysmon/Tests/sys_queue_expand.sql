-- Use prepared statements to force sys.queue() to expand its query circle.
-- This test relies on the little error that once a prepared statement is
--   executed, the "prepare ..." query is not removed from sys.queue()
prepare select 0;
prepare select 1;
prepare select 2;
prepare select 3;
prepare select 4;
prepare select 5;
prepare select 6;
prepare select 7;
select username, status, query from sys.queue();
exec 0();
select 0, username, status, query from sys.queue();
exec 1();
select 1, username, status, query from sys.queue();
exec 2();
select 2, username, status, query from sys.queue();
exec 3();
select 3, username, status, query from sys.queue();
exec 4();
select 4, username, status, query from sys.queue();
exec 5();
exec 6();
exec 7();
select 7, username, status, query from sys.queue();
select 8, username, status, query from sys.queue();
