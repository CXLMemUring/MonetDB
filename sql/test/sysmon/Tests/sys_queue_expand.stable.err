stderr of test 'sys_queue_expand` in directory 'sql/test/sysmon` itself:


# 18:07:00 >  
# 18:07:00 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-31724" "--port=38231"
# 18:07:00 >  


# 18:07:00 >  
# 18:07:00 >  "Done."
# 18:07:00 >  

