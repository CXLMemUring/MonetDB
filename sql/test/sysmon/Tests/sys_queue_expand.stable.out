stdout of test 'sys_queue_expand` in directory 'sql/test/sysmon` itself:


# 18:07:00 >  
# 18:07:00 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-31724" "--port=38231"
# 18:07:00 >  

Before sleep: 3
('before', 'monetdb', 'finished', 'select 5, username, status, query from sys.queue();')
('before', 'monetdb', 'finished', 'select 6, username, status, query from sys.queue();')
('before', 'monetdb', 'running', "select \\'before\\', username,status,query from sys.queue() order by status, query\n;")

During sleep: 4
('during', 'monetdb', 'running', 'call sleep(3000)\n;')
('during', 'monetdb', 'running', 'call sleep(3000)\n;')
('during', 'monetdb', 'running', 'call sleep(3000)\n;')
('during', 'monetdb', 'running', "select \\'during\\', username,status,query from sys.queue() order by status, query\n;")

After sleep: 5
('after', 'monetdb', 'finished', 'call sleep(3000)\n;')
('after', 'monetdb', 'finished', 'call sleep(3000)\n;')
('after', 'monetdb', 'finished', 'call sleep(3000)\n;')
('after', 'monetdb', 'finished', "select \\'during\\', username,status,query from sys.queue() order by status, query\n;")
('after', 'monetdb', 'running', "select \\'after\\', username,status,query from sys.queue() order by status, query\n;")

# 16:54:22 >  
# 16:54:22 >  "Done."
# 16:54:22 >  

